utils/common/vlarray.h

```diff
@@ -0,0 +1,81 @@
+#ifndef UTILS_COMMON_VLARRAY_H
+#define UTILS_COMMON_VLARRAY_H
+
+#include <boost/noncopyable.hpp>
+
+namespace utils {
+
+template <typename T, size_t SIZE=256>
+class VLArray : boost::noncopyable
```

Comment:
Very nice approach. 
What do you think if we reduce boost importance here and employ an approach like [this](https://www.codeproject.com/Tips/1220451/Cplusplus-Tip-Make-Your-Class-Non-copyable-Without)? Will this work or not?